File: go/excel-sheet-column-title.go
func convertToTitle(columnNumber int) string {
	A := int('A')
	title := make([]byte, 0)
	for 0 < columnNumber {
		charInt := columnNumber%26 - 1
		if charInt < 0 {
			columnNumber -= 1
			charInt = 25
		}
		char := byte(charInt + A)
		title = append([]byte{char}, title...)
		columnNumber /= 26
	}
	return string(title)
}

File: python/excel-sheet-column-title.py
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        title_array = []
        while 0 < columnNumber:
            c_int = columnNumber % 26 - 1
            if c_int == -1:
                c_int = 25
            c = chr(c_int + ord("A"))
            title_array.append(c)
            if c_int == 25:
                columnNumber -= 1
            columnNumber //= 26
        title_array.reverse()
        return "".join(title_array)
